deploy/ui/form.json

```
@@ -107,26 +107,63 @@
           ]
         },
         {
-          "label": "Admin user",
-          "type": "fieldGroup",
+          "label": "Admin user configuration",
+          "type": "dropDown",
           "required": false,
           "jsonPath": "$.spec.commonConfig",
+          "options": [
+            {
+              "value": "secret",
+              "label": "Secret configuration"
+            },
+            {
+              "value": "default",
+              "label": "Username and Password configuration"
+            }
+          ],
           "visible": true,
           "fields": [
             {
-              "label": "Username",
-              "default": "adminUser",
-              "required": false,
-              "description": "Administrative username for RHPAM components",
-              "jsonPath": "$.spec.commonConfig.adminUser",
-              "type": "text"
+              "label": "Secret to use",
+              "type": "fieldGroup",
+              "required": true,
+              "jsonPath": "$.spec.commonConfig",
+              "displayWhen": "secret",
+              "visible": true,
+              "fields": [
+                {
+                  "label": "Secret configuration to use for Admin user",
+                  "type": "text",
+                  "jsonPath": "$.spec.commonConfig.secretAdminCredentials",
+                  "originalJsonPath": "$.spec.commonConfig.secretAdminCredentials",
+                  "description": "If empty kie-admin-credentials secret will be used, if missing one with default username and generated password will be created"
+                }
+              ]
             },
             {
-              "label": "Password",
-              "required": false,
-              "description": "Administrative password to use instead of generating default value",
-              "jsonPath": "$.spec.commonConfig.adminPassword",
-              "type": "password"
+              "label": "Username and password to create a kie-admin-credentials secret",
```

Will be really credentials secret created? In this case this can be breaking change and we probably don't wont to do that. AdminUser and AdminPassword should be still creating the user in the old way, without the secret.
```suggestion
              "label": "Username and password to create",
```
Or is this just some typo or copy paste leftover?

---

This is the new behaviour decided with Felipe, no more username and password in clear text but in the secrets